app/notification.go

```diff
@@ -731,12 +720,25 @@ func (a *App) sendPushNotification(post *model.Post, user *model.User, channel *
 	msg.RootId = post.RootId
 	msg.SenderId = post.UserId
 
+	if !sentBySystem {
+		senderName = sender.GetDisplayName(teammateNameConfig)
+		preference, prefError := a.GetPreferenceByCategoryAndNameForUser(user.Id, model.PREFERENCE_CATEGORY_DISPLAY_SETTINGS, "name_format")
+		if prefError == nil && preference.Value != teammateNameConfig {
+			senderName = sender.GetDisplayName(preference.Value)
+		}
+	}
+
+	if channel.Type == model.CHANNEL_DIRECT {
+		channelName = fmt.Sprintf("@%v", senderName)
+	}
+
 	if contentsConfig != model.GENERIC_NO_CHANNEL_NOTIFICATION || channel.Type == model.CHANNEL_DIRECT {
 		msg.ChannelName = channelName
 	}
 
 	if ou, ok := post.Props["override_username"].(string); ok {
 		msg.OverrideUsername = ou
+		senderName = ou
```

Comment:
Do we need to check if we have username overrides enabled here, or is that done prior to this?

Reply:
I've added the check just to be sure, for the username and the icon